components/NavigationDrawer/src/MDCBottomDrawerPresentationController.m

```diff
@@ -0,0 +1,146 @@
+  if ([self.presentedViewController isKindOfClass:[MDCBottomDrawerViewController class]]) {
+    MDCBottomDrawerViewController *bottomDrawerViewController =
+        (MDCBottomDrawerViewController *)self.presentedViewController;
+    bottomDrawerContainerViewController.contentViewController =
+        bottomDrawerViewController.contentViewController;
+    bottomDrawerContainerViewController.headerViewController =
+        bottomDrawerViewController.headerViewController;
+  } else {
+    bottomDrawerContainerViewController.contentViewController = self.presentedViewController;
+  }
+  bottomDrawerContainerViewController.animatingPresentation = YES;
+  self.bottomDrawerContainerViewController = bottomDrawerContainerViewController;
+
+  self.scrimView = [[UIView alloc] initWithFrame:self.containerView.bounds];
+  self.scrimView.backgroundColor = MDCBottomDrawerOverlayBackgroundColor();
+  self.scrimView.autoresizingMask =
+      UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+  self.scrimView.accessibilityIdentifier = @"Close drawer";
+  self.scrimView.accessibilityTraits |= UIAccessibilityTraitButton;
+
+  [containerView addSubview:self.scrimView];
+  [containerView addSubview:self.bottomDrawerContainerViewController.view];
+
+  // Set up the tap recognizer to dimiss the drawer by.
+  UITapGestureRecognizer *tapGestureRecognizer =
+      [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(hideDrawer)];
+  [containerView addGestureRecognizer:tapGestureRecognizer];
+  tapGestureRecognizer.delegate = self;
+
+  id<UIViewControllerTransitionCoordinator> transitionCoordinator =
+      [[self presentingViewController] transitionCoordinator];
+
+  // Fade in the scrim view during the transition.
+  self.scrimView.alpha = 0.0;
+  [transitionCoordinator
+      animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {
+        self.scrimView.alpha = 1.0;
+      }
+                      completion:nil];
+}
+
+- (void)presentationTransitionDidEnd:(BOOL)completed {
+  self.bottomDrawerContainerViewController.animatingPresentation = NO;
+  [self.bottomDrawerContainerViewController.view setNeedsLayout];
+  if (!completed) {
+    [self.scrimView removeFromSuperview];
+  }
+}
+
+- (void)dismissalTransitionWillBegin {
+  id<UIViewControllerTransitionCoordinator> transitionCoordinator =
+      [[self presentingViewController] transitionCoordinator];
+  [transitionCoordinator
+      animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext> context) {
+        self.scrimView.alpha = 0.0;
+      }
+                      completion:nil];
+}
+
+- (void)dismissalTransitionDidEnd:(BOOL)completed {
+  if (completed) {
+    [self.scrimView removeFromSuperview];
+  }
+}
+
+- (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentContainer>)container {
+  [super preferredContentSizeDidChangeForChildContentContainer:container];
```

Comment:
I can confirm what yarneo mentioned in the above comment - this is just setting grounds for upcoming support.
I do think that it will be useful to add an example when the presented view controller's preferredContentSize is smaller than the screen, like ianegordon suggested. I can verify the expected behavior is to have the height be respected while eh width get overridden to fit the with of the presenting view.

Reply:
As this is only setting the ground for dynamically changing content, let's punt adding another dragon as a follow up to landing the component itself.